src/sidebar/oauth-auth.js

```diff
@@ -223,17 +215,25 @@ function auth($http, $rootScope, $window,
    */
   function tokenGetter() {
     if (!tokenInfoPromise) {
-      var grantToken = grantTokenFromHostPage();
-
-      if (grantToken) {
-        // Exchange host-page provided grant token for a new access token.
-        tokenInfoPromise = exchangeJWT(grantToken).then((tokenInfo) => {
-          return tokenInfo;
-        }).catch(function(err) {
-          showAccessTokenExpiredErrorMessage(
-            'You must reload the page to annotate.');
-          throw err;
-        });
+      var cfg = serviceConfig(settings);
+
+      // Check if automatic login is being used, indicated by the presence of
+      // the 'grantToken' property in the service configuration.
+      if (cfg && typeof cfg.grantToken !== 'undefined') {
+        if (cfg.grantToken) {
```

Comment:
As [discussed in Slack](https://hypothes-is.slack.com/archives/C4K6M7P5E/p1510596981000246?thread_ts=1510583846.000041&cid=C4K6M7P5E) there's an unhandled case here when there _is_ a service config but it _doesn't_ contain any `grantToken` (not even `null`). This should be fixed elsewhere, in a separate PR: a service config without a `grantToken` should be considered invalid and rejected. The docs for `grantToken` need to be updated to say that it's required and does _not_ default to `null`.